### PR TITLE
Custom asset timestamp

### PR DESCRIPTION
**Pull request by fquffio**
This PR implements new algorithms to generate hashes to be appended to static resources URLs (see #1039).

By specifying `$config['Asset']['algorithm']` one can choose between:
- `md5` and `md5-short`: calculates MD5 of static asset (or only first 7 characters).
- `sha1` and `sha1-short`: same as above, with SHA1 algorithm.
- `git` and `git-short`: full or short hash commit of Git repository to which file belongs.
- _default_: last modification timestamp.

### CODE REVIEW

File: bedita-app/views/helpers/be_html.php [hunk 1]
```diff
@@ -132,14 +144,67 @@ public function hyphen($text, $lang = null, $excludeSelectors = array('.formula'
...
         return $text;
     }
 
+    /**
+     * Append timestamp to an asset to help invalidate browsers' cache.
+     *
+     * @param string $path File path.
+     * @return string
+     */
+    public function assetTimestamp($path) {
```
Comment (stefanorosanelli) [inline bedita-app/views/helpers/be_html.php:163]: Maybe `assetVersion` or `assetRevision`? Timestamp is one of the options

**Review by stefanorosanelli — COMMENTED** [review]
Code seems ok (apart naming maybe)

How are we going to use this helper method in a template?
Should we maybe introduce `$beHtml->css()` or `$beHtml->script()` ?

File: bedita-app/views/helpers/be_html.php (hunk 1 above)
Comment (fquffio) [reply bedita-app/views/helpers/be_html.php:163]: @stefanorosanelli Sorry, I read your comment just now. Don't know why GitHub notifications aren't working for me. 😕 

Anyway: we don't need `BeHtmlHelper::css()` nor `BeHtmlHelper::script()`: `BeHtmlHelper` extends CakePHP's `HtmlHelper` that implements those methods. Such methods internally invoke `HtmlHelper::assetTimestamp()`, that is overridden by this PR.

Long story short: `$beHtml->script()` and `$beHtml->css()` do the work just fine. If we rename `BeHtmlHelper::assetTimestamp()` to something that fits more the scope of this PR we need to make the magic happen again by overriding `HtmlHelper::css()` and `HtmlHelper::script()` as well.

Comment (stefanorosanelli) [reply bedita-app/views/helpers/be_html.php:163]: ok, didn't notice method override

**Pull request merged**